lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,344 @@
+    if (!(evt.relatedTarget instanceof Node && evt.currentTarget.contains(evt.relatedTarget))) {
+      setShowDropdown(false);
+    }
+  }
+
+  function handleFilterChange(value: string) {
+    setElToFocusId('');
+    setFocusableBtnIndex(null);
+    onSearchTextChange(value);
+
+    if (value && value.trim() !== searchText.trim()) {
+      doSearch(value);
+      setPrevSearchText('');
+    }
+    else if (!value) {
+      setShowDropdown(false);
+    }
+  }
+
+  // helper for focusing different buttons in the dropdown menu
+  const adjustBtnFocus = (adjust: (i: number) => number) => () => {
+        const newFocusableBtnIndex = adjust(focusableBtnIndex ?? 0),
+            elToFocus = dropdownRef.current?.children[newFocusableBtnIndex] as HTMLElement | null;
+        if (elToFocus) {
+          setElToFocusId(elToFocus.id);
+          setFocusableBtnIndex(newFocusableBtnIndex);
+          if (autoComplete && typeof elToFocus.textContent === 'string') {
+            onSearchTextChange(elToFocus.textContent);
+          }
+        }
+      },
+      focusNext = adjustBtnFocus(inc),
+      focusPrev = adjustBtnFocus(dec),
+      focusFirst = adjustBtnFocus(always(0)),
+      focusLast = adjustBtnFocus(always(matches.length - 1));
+
+  function handleKeyDown(evt: KeyboardEvent<HTMLElement>) {
+    const inputEle = evt.currentTarget as HTMLInputElement;
+    switch (evt.key) {
+      case 'Enter':
+        if (typeof focusableBtnIndex === 'number' && elToFocusId) {
+          handleOnClick(matches[focusableBtnIndex]);
+        }
+        evt.preventDefault();
+        break;
+      case 'Home':
+        inputEle.setSelectionRange(0, 0);
+        evt.preventDefault();
+        break;
+      case 'End':
+        inputEle.setSelectionRange(searchText.length, searchText.length);
+        evt.preventDefault();
+        break;
+      case 'ArrowDown':
+        if (!isEmpty) {
+          setShowDropdown(true);
+        }
+        elToFocusId
+          ? (dropdownRef.current?.children[matches.length - 1].id === elToFocusId ? focusFirst() : focusNext())
+          : focusFirst();
+        evt.preventDefault();
+        break;
+      case 'ArrowUp':
+        if (!isEmpty) {
+          setShowDropdown(true);
+        }
+        elToFocusId ? (dropdownRef.current?.children[0].id === elToFocusId ? focusLast() : focusPrev()) :
+          focusLast();
+        evt.preventDefault();
+        break;
+      case 'Escape':
+        setShowDropdown(false);
+        setElToFocusId('');
+        setFocusableBtnIndex(null);
+        if (!showDropdown) {
+          handleFilterChange('');
+        }
+        break;
+    }
+  }
+
+  function doSearch(value: string) {
+    onSearch(value.trim());
+    setShowDropdown(true);
+  }
+
+  function focusTextInput() {
+    inputRef.current?.querySelector('input')?.focus();
+  }
+
+  function setInlineOption() {
+    if (typeof matches[0].displayName === 'string' &&
```

Comment:
I imagine this component can't really work with `DataItem`s that have a `displayName` that is anything other than a string.  We should probably add another type parameter to `DataItem`'s definition which allows us to specify that this component only accepts `DataItem`s that use `string` `displayName`s

Reply:
Do you mean adding new interface in /util/DataItem like below?
`export interface DataItemWithStringName<T extends string | number = string> {`
 ` id: T;`
  `displayName: string | null;`
`}`

Reply:
> Do you mean adding new interface in /util/DataItem like below?

No, I mean adjusting the existing `DataItem` to be like this:
```
type RequiredReactNode = Exclude<ReactNode, null | undefined>;
export interface DataItem<I extends string | number = string, D extends RequiredReactNode = RequiredReactNode> {
  id: I;
  displayName: D;
}
```

Reply:
> No, I mean adjusting the existing DataItem to be like this:

I'm hoping that by having a default for the second type parameter, that all the existing code that uses a single type parameter will still work.  If that turns out not to be the case, then we won't be able to go this way as it'll be a breaking change.

Reply:
I implemented your suggestion above and other codes using a single type are still working, now I could have `string` type specified for `displayName`.
Updated here d99911ec4323048a2c2273180b1cbe2151ebf6da. Thanks.